### PR TITLE
Fix build issues

### PR DESCRIPTION
**Pull request by edgchen1**
Fixed various compilation issues. Still need to address test failures.

### CODE REVIEW

File: onnxruntime/core/framework/session_state.cc [hunk 1]
```diff
@@ -254,7 +254,7 @@ Status SessionState::GeneratePatternGroupCache(const std::vector<std::reference_
             len *= dim.dim_value();
           }
         }
-        if (!IAllocator::CalcMemSizeForArrayWithAlignment<64>(len, ml_data_type->Size(), &size)) {
+        if (!IAllocator::CalcMemSizeForArrayWithAlignment<64>(static_cast<size_t>(len), ml_data_type->Size(), &size)) {
```
Comment (edgchen1) [inline onnxruntime/core/framework/session_state.cc]: i wonder if we should use SafeInt here, i think that was introduced recently and is intended for allocation related calculations

Comment (snnn) [reply onnxruntime/core/framework/session_state.cc]: Please.

**Comment by SherlockNoMad** [comment]
Looks good to me.

**Pull request merged**